### PR TITLE
Cleaned up memcached example and added tests.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and applications on [Google App Engine](http://cloud.google.com/nodejs).
 ### Frameworks
 
 - Express.js - [Source code][express_1] | [App Engine Tutorial][express_2] | [Live demo][express_3] | [Documentation][express_4]
+  - Express.js + Memcached Sessions - [Source code][express_5] | [Documentation][express_6]
```
Comment (jonparrott) [inline README.md]: nit that you don't have to fix now: it would be better to have more described anchors here. `express_6` is kinda of funny.

File: appengine/express-memcached-session/README.md [hunk 2]
```diff
@@ -1,10 +1,14 @@
-# Express + Memcached Sessions -> Google App Engine
+## Express.js + Memcached Sessions on Google App Engine
```
Comment (jonparrott) [inline appengine/express-memcached-session/README.md]: Seems like this readme is too heavy. What's the point of the sample code if you put everything in the readme? :P

File: appengine/express-memcached-session/server.js [hunk 3]
```diff
@@ -24,29 +24,33 @@ var app = express();
...
 app.use(cookieParser());
 app.use(session({
-    secret: 'appengineFTW',
-    key: 'test',
-    proxy: 'true',
-    store: new MemcachedStore({
-        hosts: [process.env.MEMCACHE_URL || '127.0.0.1:11211']
-    })
+  secret: 'your-secret-here',
+  key: 'test',
```
Comment (jonparrott) [inline appengine/express-memcached-session/server.js]: Surely we can pick a better default key.

File: appengine/express-memcached-session/server.js [hunk 4]
```diff
@@ -24,29 +24,33 @@ var app = express();
...
+    } else {
+      req.session.views = 1;
+    }
+    res.end('Viewed <strong>' + req.session.views + '</strong> times.');
+  });
 });
 
-http.createServer(app).listen(process.env.PORT || 8080, function() {
+if (module === require.main) {
+  http.createServer(app).listen(process.env.PORT || 8080, function() {
```
Comment (jonparrott) [inline appengine/express-memcached-session/server.js]: why not just `app.listen`?

File: appengine/express-memcached-session/server.js [hunk 5]
```diff
@@ -24,29 +24,33 @@ var app = express();
...
+      ++req.session.views;
+    } else {
+      req.session.views = 1;
+    }
+    res.end('Viewed <strong>' + req.session.views + '</strong> times.');
+  });
 });
 
-http.createServer(app).listen(process.env.PORT || 8080, function() {
+if (module === require.main) {
```
Comment (jonparrott) [inline appengine/express-memcached-session/server.js]: why is this needed here? Is it for tests? I don't often see this in other samples.

File: test/appengine/express-memcached-session.test.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+var request = require('supertest');
+var app = require('../../appengine/express-memcached-session/server.js');
+
+describe('express-memcached-session', function () {
+  it('should return 200 and the correct response', function (done) {
+    request(app)
+      .get('/')
+      .expect(200)
+      .expect(function (res) {
+        assert(res.text.match(/<div>(\d|\d\d|\d\d\d)\.(\d|\d\d|\d\d\d)\.(\d|\d\d|\d\d\d)\.(\d|\d\d|\d\d\d)<\/div>Viewed <strong>\d<\/strong> times./), // jshint ignore:line
```
Comment (jonparrott) [inline test/appengine/express-memcached-session.test.js]: Regexs are like violence, if they aren't working, you aren't using enough of it.

**Comment by JustinBeckwith** [comment]
LGTM.

File: appengine/express-memcached-session/README.md (hunk 2 above)
Comment (jmdobry) [inline appengine/express-memcached-session/README.md]: I made the readme more lightweight, it now mirrors more closely the format used by the tutorials for the other framework examples.

File: appengine/express-memcached-session/server.js (hunk 5 above)
Comment (jmdobry) [inline appengine/express-memcached-session/server.js]: for tests

**Comment by jmdobry** [comment]
I simplified the test and the readme. Should be good to go.

File: appengine/express-memcached-session/server.js [hunk 7]
```diff
@@ -24,29 +24,33 @@ var app = express();
...
-    key: 'test',
-    proxy: 'true',
-    store: new MemcachedStore({
-        hosts: [process.env.MEMCACHE_URL || '127.0.0.1:11211']
-    })
+  secret: 'your-secret-here',
+  key: 'view:count',
+  proxy: 'true',
+  store: new MemcachedStore({
+    hosts: [process.env.MEMCACHE_URL || '127.0.0.1:11211']
```
Comment (jonparrott) [inline appengine/express-memcached-session/server.js]: Now that you slimmed down the readme, a little comment here about how `MEMCACHE_URL` is set by GAE would be helpful. :)

File: appengine/express-memcached-session/server.js (hunk 5 above)
Comment (jonparrott) [inline appengine/express-memcached-session/server.js]: gotcha.

**Comment by jonparrott** [comment]
LGTM w/ final nit.

File: appengine/express-memcached-session/server.js (hunk 7 above)
Comment (jmdobry) [inline appengine/express-memcached-session/server.js]: It's set by GAE? So it doesn't need to be set in the `app.yaml` file?

Comment (jonparrott) [inline appengine/express-memcached-session/server.js]: Nvm, just saw the full thing. A comment in the readme about the `memcache` host then. :P

Comment (jmdobry) [inline appengine/express-memcached-session/server.js]: I think there is one at the bottom.
